crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -62,6 +66,111 @@ where
+        let mut at = block.parent_hash;
+        let mut replay_block_txs = true;
+
+        // but if all transactions are to be replayed, we can use the state at the block itself
+        let num_txs = transaction_index.index().unwrap_or(block.body.len());
+        if num_txs == block.body.len() {
+            at = block.hash;
+            replay_block_txs = false;
+        }
+
+        let this = self.clone();
+
+        self.inner
+            .tracing_call_pool
+            .spawn(move || {
+                let state = this.state_at(at.into())?;
+
+                let mut results: Vec<EthCallResponse> = Vec::with_capacity(transactions.len());
+                let mut db = SubState::new(State::new(state));
+
+                if replay_block_txs {
+                    // only need to replay the transactions in the block if not all transactions are
+                    // to be replayed
+                    let transactions = block.body.into_iter().take(num_txs);
+
+                    // Execute all transactions until index
+                    for tx in transactions {
+                        let tx = tx.into_ecrecovered().ok_or(BlockError::InvalidSignature)?;
+                        let tx = tx_env_with_recovered(&tx);
+                        let env = Env { cfg: cfg.clone(), block: block_env.clone(), tx };
+                        let (res, _) = transact(&mut db, env)?;
+                        db.commit(res.state);
+                    }
+                }
+
+                let overrides =
+                    EvmOverrides::new(state_override.clone(), block_override.map(Box::new));
+
+                let transactions = transactions.into_iter().peekable();
+                for tx in transactions {
+                    let env = prepare_call_env(
+                        cfg.clone(),
+                        block_env.clone(),
+                        tx,
+                        gas_limit,
+                        &mut db,
+                        overrides.clone(),
+                    )?;
+                    let (res, _) = transact(&mut db, env)?;
+
+                    match res.result {
+                        ExecutionResult::Success { output, .. } => {
+                            results.push(EthCallResponse {
+                                output: Some(output.into_data().into()),
+                                error: None,
+                            });
+                        }
+                        ExecutionResult::Revert { output, .. } => {
+                            results
+                                .push(EthCallResponse { output: None, error: Some(output.into()) });
+                        }
+                        ExecutionResult::Halt { .. } => {
+                            results.push(EthCallResponse {
+                                output: None,
+                                error: Some(Default::default()),
+                            });
+                        }
```

Comment:
see comment re error field, 

if error is a string then we can use:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/error.rs#L537-L547